### ISSUE TITLE
Gridline-, pixel-, or both registrations?

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmtserver-admin#35]
The earth_relief_xxy DEMs we supply have different registrations: The SRTM 1s are pixel registered, as is the 15s SRTM15+v2.  For 30s and up we downsample the 15s grid and write gridline registered files.  This was partly motivated by the registrations of past DEMs (like ETOPO1m) as well as wanting to match the registrations of other useful grids (e.g., crustal ages).  However, as we add more remote grids and especially images, there will inevitably be a mismatch, and then we are forcing the user to do painful things like resample one item from pixel to grid or vice versa, loosing the highest frequencies in the process.

Another solution is to supply both **g** and **p** versions on the downsampled versions. Thus, while 1s, 3s, 15s must remain in their original pixel registration, 30s and up could come in two flavors, e.g.

`earth_relief_01m[p|g]`

The way this would work is that if no **p**|**g** are indicated then we get the default registration [TBD], while if you are specific you get that particular version, if it exists, else you get a warning and we serve you the other one.  Most users won't care or understand the difference, but people who need to get a 1-minute DEM that matches the pixel registration on some global geotiff may ask for the pixel-registered DEM.  I would only apply this scheme to geophysical data, not images (which thus will be pixel only).

I am contemplating this for my own needs as well: I do make maps of crustal ages using shading from DEMs, and I do analyze depth-age from the two grids, and it is always super-annoying if I need to switch registration as I loose information.  And of course, all the global geotiffs are pixel registered and thus I rather use a pixel-registered 01m DEM than fuss with the conversion myself.

If someone naively uses grdgradient on the gridline DEM then uses that as intensities with a pixel image then we can give a helpful error message when the registration clashes.

I wish I could automate the selection but it seems difficult to know what the user needs ahead of time.  Thoughts from @GenericMappingTools/gmt-contributors ?

**Comment by PaulWessel** [comment]
@WalterHFSmith, you can see in detail how I downsample the DEM by looking at the srv_downsampler.sh script in the scripts folder - visible under code.  That script is fed the earth_relief.recipe file (in the recipe directory) and off it goes.  I will have to add a loop over g and p resolutions.

**Comment by WalterHFSmith** [comment]
Is it possible to figure out from the user's plot what might be wanted? E.g. a cylindrical map projection might want a pixel-registered data file...

Downsampling means different things to different people, and best way depends on application, perhaps. For example, a downsample via block median might preserve the above/below zero -ness of the data, which might look right if the zero contour or land/sea color was important.

Also let us remember that merely moving between pixel and grid, without intending any down-sampling at all, still reduces amplitude by 50% at a wavelength of 10 delta_x. See attached Figure 11 from DOI 10.1007/s11001-005-2095-4

[Fig11.pdf](https://github.com/GenericMappingTools/gmtserver-admin/files/4685457/Fig11.pdf)

**Comment by PaulWessel** [comment]
Your Fig 11 is why I added both pixel and grid for the filtered products.  So that if a user has a product that is 5x5 minute in pixel they would not have to take ETOPO5m and "resample" to pixel and thus lose information.
I thought a bit about auto-selecting g or p for users who don't specify.  We cannot guess too much I think.  If cylindrical or similar then perhaps p is better so we don't have to clip those half-pixels, but we do and it works fine.  I currently let gridline be default because that is the only flavor we offered in 6.0, so suddenly switching to gridline may throw some a curve.
